force-app/main/default/classes/SOQL.cls

```diff
@@ -271,91 +338,151 @@ public with sharing class SOQL {
      */
     public static List<Contact> getAccountsWithContactsByState(Id accountId) {
         // Query the Account
-
+        String billingState = [SELECT BillingState FROM Account Where ID = :accountId].BillingState;
```

Comment:
Starting to look at best practices it's not recommended to return soql values into non list variables.
Try returning the accounts into a collection